projects/core/styles/components/link.less

```diff
@@ -59,13 +58,24 @@
         background: color-mix(in lch, currentColor 12%, transparent);
     });
 
-    @media @tui-mouse {
-        &[data-appearance='']:hover {
-            opacity: 0.7;
-        }
+    .appearance-hover({
+        text-decoration: underline;
+    });
+
+    [tuiTheme='dark'] &[data-appearance=''] {
+        text-decoration-color: var(--tui-service-underline-grayscale, rgba(255, 255, 255, 0.32));
+        color: var(--tui-text-primary);
+    }
+
+    &[data-appearance=''],
+    [tuiTheme='light'] &[data-appearance=''] {
+        text-decoration: underline;
+        text-decoration-color: var(--tui-service-underline-grayscale, rgba(0, 0, 0, 0.32));
```

Comment:
Do not overcomplicate this for this minor detail, use `color-mix` with `currentColor` and `transparent` like before, just update line 26 to have `solid` be default.

Then in `data-appearance="action"` make `text-decoration-line: none` so it's hidden in that appearance, and keep `.appearance-hover(...)` rule that you've added, that's all.

Reply:
fixed